modules/add_credit/src/AddCreditService.php

```diff
@@ -257,6 +260,69 @@ public function fieldWidgetFormAlter(&$element, FormStateInterface $form_state,
     }
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function formAlter(&$form, FormStateInterface $form_state, $form_id) {
+    if (($flow = $form_state->getFormObject())
+      && ($flow instanceof CheckoutFlowBase)
+      && ($form['#step_id'] == 'review')
+    ) {
+      // Add a custom validation handler to check for add credit products.
+      array_unshift($form['#validate'], [static::class, 'checkoutFormReviewValidate']);
+    }
+  }
+
+  /**
+   * Custom validation handler for the checkout review step.
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  public static function checkoutFormReviewValidate(array $form, FormStateInterface $form_state) {
+    if ($flow = $form_state->getFormObject()) {
+      // Loop through all order items.
+      // Find the add_credit products and the total amount per currency.
+      /** @var \Drupal\commerce_order\Entity\OrderItemInterface $item */
+      $add_credit_totals = [];
+      foreach ($flow->getOrder()->getItems() as $item) {
+        /** @var \Drupal\commerce_product\Entity\ProductInterface $product */
+        if (($product = $item->getPurchasedEntity()->getProduct())
+          && ($product->hasField('apigee_add_credit_enabled'))
+          && ($product->get('apigee_add_credit_enabled')->value)) {
+          // Use the total price since this factors in the quantity.
+          $price = $item->getTotalPrice();
+          $currency_code = strtolower($price->getCurrencyCode());
+          $add_credit_totals[$currency_code] = !empty($add_credit_totals[$currency_code]) ? $add_credit_totals[$currency_code]->add($price) : $price;
```

Comment:
@Jaesin yeah this handles multiple line items from different accounts too.

Reply:
I see this adding all of the items with the the same total and then validating the total regardless of the `add_credit_account`.

Wouldn't it be easier to just make sure each line item is above the minimum amount.

Reply:
I think if you have 2 items that together makes the minimum amount, you should still be able to top up.

Reply:
If you are setting up an add credit product, you should be making sure a one single item in the cart meets the minimum. We wanted to have validation on that when the product is saved regardless of if they are using the price range option but this solution at least prevents checking out with a top up that won't pass the check in the edge API. `\Drupal\apigee_m10n_add_credit\EventSubscriber\CommerceOrderTransitionSubscriber::handleOrderStateChange` does aggregate the amout before adding the credit to the account but that is just to reduce the number of calls to the API endpoint for one single order. The method also considers the `add_credit_account` data when calculating the totals.

We could try to consider all of the other aspects of the line item in this validaiton but the easiest solution is to validate per line item since the commerce product is supposed to have the minimum amount validated anyway.

Reply:
Done.